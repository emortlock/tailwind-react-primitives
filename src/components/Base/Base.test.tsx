import React from 'react'
import { render, RenderResult } from '@testing-library/react'

import { Base } from '.'

const setup = (
  testProps = {},
): [RenderResult, { props: Record<string, any> }] => {
  const props = Object.assign({}, testProps)

  const wrapper = render(<Base {...props}>Hello World</Base>)

  return [
    wrapper,
    {
      props,
    },
  ]
}

describe('Base', () => {
  it('renders matching snapshot', () => {
    const [{ asFragment }] = setup({ focusable: true })

    expect(asFragment()).toMatchSnapshot()
  })
})
