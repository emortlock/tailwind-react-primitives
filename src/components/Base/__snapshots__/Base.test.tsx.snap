// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base renders matching snapshot 1`] = `
<DocumentFragment>
  <div
    class="focus:outline-none focus:shadow-outline"
  >
    Hello World
  </div>
</DocumentFragment>
`;
